src/jit/compiler.cpp

```diff
@@ -11097,3 +11097,831 @@ bool Compiler::killGCRefs(GenTree* tree)
+        {
+            dumpVariableLiveRange(&it, _emitter, codeGen);
+        }
+        printf("]\n");
+    }
+}
+
+void VariableLiveDescriptor::dumpRegisterLiveRangesForBlockBeforeCodeGenerated(const CodeGenInterface* codeGen) const
+{
+    // "m_VariableLifeBarrier" should has been initialized
+    noway_assert(m_VariableLifeBarrier != nullptr);
+
+    if (hasVarHomesToDump())
+    {
+        printf("[");
+        for (LiveRangeListIterator it = m_VariableLifeBarrier->getStartForDump(); it != m_VariableLiveRanges->end();
+             it++)
+        {
+            dumpVariableLiveRange(&it, codeGen);
+        }
+        printf("]\n");
+    }
+    else
+    {
+        printf("None history\n");
+    }
+}
+
+// Returns true if a live range for this variable has been recorded from last call to EndBlock
+bool VariableLiveDescriptor::hasVarHomesToDump() const
+{
+    // "m_VariableLifeBarrier" should has been initialized
+    noway_assert(m_VariableLifeBarrier != nullptr);
+
+    return m_VariableLifeBarrier->hasLiveRangesToDump();
+}
+
+// Reset the barrier so as to dump only next block changes on next block
+void VariableLiveDescriptor::endBlockLiveRanges()
+{
+    noway_assert(m_VariableLifeBarrier != nullptr);
+
+    // make "m_VariableLifeBarrier->m_StartingLiveRange" now points to nullptr for printing purposes
+    m_VariableLifeBarrier->resetDumper(m_VariableLiveRanges);
+}
+
+#endif // DEBUG
+
+//------------------------------------------------------------------------
+//                      VariableLiveKeeper
+//------------------------------------------------------------------------
+
+void Compiler::initializeVariableLiveKeeper()
+{
+    CompAllocator allocator = getAllocator(CMK_VariableLiveRanges);
+
+    int amountTrackedVariables = opts.compDbgInfo ? info.compLocalsCount : 0;
+    int amountTrackedArgs      = opts.compDbgInfo ? info.compArgsCount : 0;
+
+    varLiveKeeper = allocator.allocate<VariableLiveKeeper>(1);
+    memset(varLiveKeeper, 0, sizeof(*varLiveKeeper));
+    new (varLiveKeeper, jitstd::placement_t()) VariableLiveKeeper(amountTrackedVariables, amountTrackedArgs, this);
+}
+
+VariableLiveKeeper* Compiler::getVariableLiveKeeper() const
+{
+    return varLiveKeeper;
+};
+
+//------------------------------------------------------------------------
+// VariableLiveKeeper: Create an instance of the object in charge of managing
+//  VariableLiveRanges and intialize the array "lvaLiveDsc".
+//
+// Arguments:
+//    totalLocalCount   - the amount of args, special args and IL Local
+//      variables in the method.
+//    argsCount         - the amount of args and special args in the method.
+//    compiler          - a compiler instance
+//
+VariableLiveKeeper::VariableLiveKeeper(unsigned int totalLocalCount, unsigned int argsCount, Compiler* comp)
+{
+    // Fill internal properties
+    lvLiveDscCount              = totalLocalCount;
```

Comment:
Can't these use the ` : x(_x), y(_y)` style ctor syntax?

Reply:
I like what you say, thanks!